postgres_composite_types/operations.py

```diff
@@ -25,28 +27,32 @@ def sql_drop_type(type_name, schema_editor):
     return f"DROP TYPE {quoted_name}"
 
 
-class BaseOperation(Operation):
+class CreateType(CreateModel):
     """Base class for the DB operation that relates to this type."""
 
     reversible = True
-    Meta = None
 
-    def state_forwards(self, app_label, state):
-        pass
+    def __init__(self, *, name: str, fields, options) -> None:
+        fields = [
+            (DummyField.name, DummyField(primary_key=True, serialize=False)),
+            *fields,
+        ]
+        super().__init__(name, fields, options)
 
     def describe(self):
-        return f"Creates type {self.Meta.db_type}"
+        return f"Creates type {self.name}"
+
+    def state_forwards(self, app_label, state) -> None:
+        state.add_model(
+            ModelState(app_label, self.name, list(self.fields), dict(self.options))
+        )
 
     def database_forwards(self, app_label, schema_editor, from_state, to_state):
         schema_editor.execute(
-            sql_create_type(self.Meta.db_type, self.Meta.fields, schema_editor)
-        )
-        self.Meta.model.register_composite(schema_editor.connection)
-        composite_type_created.send(
```

Comment:
Just because?

Reply:
I culdn't get this to work at all. I am not sure what benefit it actually brings to the table, though.